### PR TITLE
Migrate from bintray to maven central/github packages

### PR DESCRIPTION
**Pull request by infeo**
This PR migrates the repository CI system from bintray to github packages and maven central.

This is basically copy-paste from cryptomator/integrations-win repository. Unfortunately i don't know which env/repository variables are already set and which not. This should be checked and corrected before merging this PR.

As a side note: In the notification of the github packages workflow was removed in integrations-win (see https://github.com/cryptomator/integrations-win/commit/a5640dfccdc9ff6ad2274bad2829e79593062172). If this is intended. it should also be removed here.

### CODE REVIEW

File: .github/workflows/publish-github.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Slack Notification
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_USERNAME: 'Cryptobot'
+          SLACK_ICON:
+          SLACK_ICON_EMOJI: ':bot:'
+          SLACK_CHANNEL: 'cryptomator-desktop'
+          SLACK_TITLE: "Published ${{ github.event.repository.name }} ${{ github.event.release.tag_name }}"
+          SLACK_MESSAGE: "Ready to <https://github.com/${{ github.repository }}/actions?query=workflow%3A%22Publish+to+Maven+Central%22|deploy to Maven Central>."
```
Comment (overheadhunter) [inline .github/workflows/publish-github.yml:38]: as we noticed today, the link can be changed to `https://github.com/${{ github.repository }}/actions/workflows/publish-central.yml`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,51 +1,40 @@
...
-          echo "::set-output name=version::${v}"
         shell: bash
+        run: |
+          mvn versions:set --file ./pom.xml -DnewVersion=${GITHUB_REF##*/}
       - name: Build and Test
-        run: mvn -B install
-      - name: Upload .jars
-        uses: actions/upload-artifact@v2
+        id: buildAndTest
+        run: mvn -B clean install -Pdependency-check
```
Comment (overheadhunter) [inline .github/workflows/build.yml:27]:
```suggestion
        run: mvn -B clean install
```

dependency check is not configured in pom.xml

**Review by overheadhunter — CHANGES_REQUESTED** [review]
This also requires changes to the `pom.xml` (see [integrations-api commit d136c24](https://github.com/cryptomator/integrations-api/commit/d136c242a1f9b261312e1dfcc5741b647bf4552d#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8)):
* remove `<repositories>` and global `<distributionManagement>`
* add profiles for `sign`, `deploy-central` and `deploy-github` profiles

File: .github/workflows/publish-github.yml (hunk 1 above)
Comment (infeo) [reply .github/workflows/publish-github.yml]: applied in 3187c5f

File: pom.xml [hunk 3]
```diff
@@ -292,4 +277,70 @@
...
+			</build>
+		</profile>
+
+		<profile>
+			<id>deploy-github</id>
+			<distributionManagement>
+				<repository>
+					<id>github</id>
+					<name>GitHub Packages</name>
+					<url>https://maven.pkg.github.com/cryptomator/integrations-api</url>
```
Comment (overheadhunter) [inline pom.xml:341]: wrong url

**Review by overheadhunter — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (infeo) [reply pom.xml]: Fixed in bbf36f6

**Review by overheadhunter — APPROVED** [review]

**Pull request merged**